### PR TITLE
refactor: The plugin layout state is persisted using vueuse

### PR DESCRIPTION
**Pull request by STATICHIT**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Background and solution
refactor: The plugin layout state is persisted using vueuse

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Introduced dynamic plugin layout management with new functions for retrieving and changing plugin layouts.
	- Enhanced plugin accessibility through a centralized retrieval mechanism.
	- Added alignment options for multiple plugins, improving customization of their presentation.

- **Bug Fixes**
	- Removed deprecated `addons` prop from components, streamlining plugin management.

### CODE REVIEW

**Review by lizhijie429 — APPROVED** [review]

File: packages/design-core/src/App.vue [hunk 1]
```diff
@@ -78,10 +77,14 @@ export default {
...
     addons.plugins.forEach((item) => {
-      plugin[item.id] = { width: item.options?.width || 300 }
+      if (item.id) {
+        plugin[item.id] = { width: item.options?.width || 300, align: item.options?.align || 'leftTop' }
+      }
     })
     addons.settings.forEach((item) => {
-      plugin[item.id] = { width: item.options?.width || 320 }
+      if (item.id) {
+        plugin[item.id] = { width: item.options?.width || 320, align: item.options?.align || 'leftTop' }
```
Comment (hexqi) [inline packages/design-core/src/App.vue]: 默认值应该是rightTop吧

File: packages/design-core/src/DesignPlugins.vue [hunk 2]
```diff
@@ -136,9 +132,11 @@ export default {
     const { isTemporaryPage } = usePage()
     const HELP_PLUGIN_ID = 'EditorHelp'
 
-    const { pluginState, registerPluginApi, changeLeftFixedPanels, leftFixedPanelsStorage } = useLayout()
+    const { pluginState, registerPluginApi, changeLeftFixedPanels, leftFixedPanelsStorage, getPluginLayout } =
+      useLayout()
 
-    props.addons.plugins.forEach(({ id, component, api, icon }) => {
+    const plugins = getPluginLayout('all').map((pluginName) => getPlugin(pluginName))
```
Comment (hexqi) [inline packages/design-core/src/DesignPlugins.vue]: getPluginLayout这个命名不能准确表达函数作用，用getPluginsByLayout这样会不会更好？

Comment (hexqi) [inline packages/design-core/src/DesignPlugins.vue]: 建议传’all‘改为不传参数默认返回所有

File: packages/design-core/src/DesignPlugins.vue [hunk 3]
```diff
@@ -152,9 +150,9 @@ export default {
 
     const state = reactive({
       prevIdex: -2,
-      topNavLists: plugins.filter((item) => item.align === 'top'),
-      bottomNavLists: plugins.filter((item) => item.align === 'bottom'),
-      independence: plugins.find((item) => item.align === 'independence')
+      topNavLists: getPluginLayout('leftTop').map((pluginName) => getPlugin(pluginName)),
```
Comment (hexqi) [inline packages/design-core/src/DesignPlugins.vue]: leftTop/leftBottom/... 这些字符串可以提取成常量，例如 PLUGIN_POSITION = { leftTop： 'leftTop', ... }

File: packages/controller/src/useLayout.js [hunk 4]
```diff
@@ -161,13 +161,32 @@ export default () => {
...
+
+  //获取某个布局（左上/左下/右上）的插件名称列表
+  const getPluginLayout = (layout) => {
+    const targetLayout = layout ? [] : null
+    // 遍历对象并将 align 值分类到不同的数组中
+    for (const key in pluginStorageReactive.value) {
+      if (pluginStorageReactive.value[key].align === layout || layout === 'all') {
+        targetLayout.push(key)
+      }
+    }
```
Comment (hexqi) [inline packages/controller/src/useLayout.js]: 不需要用for循环，用filter过滤就行了
```js
Object.keys(pluginStorageReactive.value).filter((key) => pluginStorageReactive.value[key].align === layout)
```

File: packages/design-core/config/addons.js [hunk 5]
```diff
@@ -65,4 +65,16 @@ const addons = {
...
+addons.plugins.forEach((item) => {
+  plugin[item.id] = item
+})
+addons.settings.forEach((item) => {
+  plugin[item.id] = item
+})
+
+export const getPlugin = (pluginName) => {
+  return plugin[pluginName] || null
+}
```
Comment (hexqi) [inline packages/design-core/config/addons.js:78]: 这段逻辑不要放到addons.js吧，抽到一个单独的文件里，addons.js里只能包含addons对象

**Review by hexqi — APPROVED** [review]

**Pull request merged**